### PR TITLE
ref(metrics): Refactor metrics stores to the new format

### PR DESCRIPTION
**Pull request by matejminar**
This PR refactors metrics stores to use `CommonStoreInterface` which allows us to consume them with the `useLegacyStore` hook. 

This will enable us to listen to the stores directly from functional components and simplify the `withMetrics*` HoCs which are used when class-based components want to listen.

### CODE REVIEW

File: tests/js/spec/stores/metricsTagStore.spec.jsx [hunk 1]
```diff
@@ -9,17 +9,17 @@ describe('MetricsTagStore', function () {
     it('should add a new tags and trigger the new addition', () => {
       jest.spyOn(MetricsTagStore, 'trigger');
 
-      const tags = MetricsTagStore.getAllTags();
-      expect(tags).toEqual({});
+      const {metricsTags} = MetricsTagStore.getState();
+      expect(metricsTags).toEqual({});
 
-      MetricsTagStore.onLoadTagsSuccess([
+      MetricsTagStore.onLoadSuccess([
```
Comment (priscilawebdev) [inline tests/js/spec/stores/metricsTagStore.spec.jsx:15]: can we have this file in typescript? 🙏

```
act(() => {
        MetricsTagStore.onLoadSuccess([
          {key: 'environment'},
          {key: 'release'},
          {key: 'session.status'},
        ]);
      });
```

if you convert it to ts, you will have to wrap this with act 😉

File: tests/js/spec/stores/metricsTagStore.spec.tsx [hunk 2]
```diff
@@ -5,21 +5,21 @@ describe('MetricsTagStore', function () {
...
     it('should add a new tags and trigger the new addition', () => {
       jest.spyOn(MetricsTagStore, 'trigger');
 
-      const tags = MetricsTagStore.getAllTags();
-      expect(tags).toEqual({});
+      const {metricsTags} = MetricsTagStore.getState();
+      expect(metricsTags).toEqual({});
 
-      MetricsTagStore.onLoadTagsSuccess([
+      MetricsTagStore.onLoadSuccess([
```
Comment (priscilawebdev) [inline tests/js/spec/stores/metricsTagStore.spec.tsx:15]: did this work without `act`?

**Review by priscilawebdev — APPROVED** [review]
thank you for converting the file to ts. it looks good to me

File: tests/js/spec/stores/metricsTagStore.spec.tsx (hunk 2 above)
Comment (matejminar) [reply tests/js/spec/stores/metricsTagStore.spec.tsx:15]: yes

Comment (priscilawebdev) [reply tests/js/spec/stores/metricsTagStore.spec.tsx:15]: nice!

**Pull request merged**